query/search/sort.go

```diff
@@ -0,0 +1,90 @@
+package search
+
+import (
+	"github.com/buger/jsonparser"
+	jsoniter "github.com/json-iterator/go"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+)
+
+// TODO: Update this to 3 once https://github.com/typesense/typesense/issues/690 is resolved
+const maxSortOrders = 2
+
+const (
+	ASC  = "$asc"
+	DESC = "$desc"
+)
+
+type Ordering = []SortField
+
+type SortField struct {
+	// Required; Name of field to enable sorting for
+	Name string
+	// Required; True if ascending order is requested, False for descending
+	Ascending bool
```

Comment:
Do you know if the cost of both ascending and descending is same? Or do you maintain some order in memory?

Reply:
I think order is generated in real time based on `text match` score for the query or `sort_by` parameter. At least, documentation does not say anything about ranking performance. 

If `default_sorting_field` is specified in collection definition, there could be some performance improvement from keeping dataset ordered in memory.